packages/react-native-eudiw-pid/src/sd-jwt/index.ts

```diff
@@ -1,65 +1,108 @@
+import base64decode from "../lib/base64";
 import { PID, Disclosure, SdJwt4VC } from "./types";
+import { pidFromToken } from "./converters";
 
 /**
- * Decode a given SD-JWT to get its readable header and payload
+ * Decode a given SD-JWT with Disclosures to get the parsed PID object they define.
+ * It ensures provided data is in a valid shape.
  *
- * @async @function
- * @param token The encoded token
+ * It DOES NOT verify token signature nor check disclosures are correctly referenced by the SD-JWT.
+ * Use {@link verify} instead
+ *
+ * @function
+ * @param token The encoded token that represents a valid sd-jwt for verifiable credentials
  *
- * @throws A decoding error
- * @returns the parsed token object
+ * @returns The validated PID object along with the parsed SD-JWT token and the parsed disclosures
+ * @throws A decoding error if the token doesn't resolve in a valid SD-JWT
+ * @throws A validation error if the provided data doesn't result in a valid PID
  *
  */
-export declare function decode(token: string): Promise<SdJwt4VC>;
+export function decode(token: string): PidWithToken {
+  // token are expected in the form "sd-jwt~disclosure0~disclosure1~...~disclosureN"
+  const [rawSdJwt, ...rawDisclosures] = token.split("~");
+
+  // get the sd-jwt as object
+  // validate it's a valid SD-JWT for Verifiable Credentials
+  const [header, payload] = rawSdJwt
+    .split(".")
+    .slice(0, 2)
+    .map(base64decode)
+    .map((e) => JSON.parse(e));
```

Comment:
Ok for now, but I expect these operations to be done by the library that handles the JWTs.

Reply:
Yes and no. I expect a chain of responsibilities:
```mermaid
graph TD;
    PID[PID: decode sdjwt+disclosures into PID] --> sdjwt[SD-JWT: decode jwt into sdjwt + decode strings into disclosures] --> jwt[JWT: decode string into jwt]
```